engine/kv_engine.cpp

```diff
@@ -881,6 +881,8 @@ Status KVEngine::batchWriteImpl(WriteBatchImpl const& batch) {
 
   // Lookup keys, allocate space according to result.
   auto ReleaseResources = [&]() {
```

Comment:
some space may be written, so we need to padding them before free

Reply:
and the log should be rollback too, otherwise the released space will be freed in rollback during recovery